pkg/apis/connector/basic_view.go

```diff
@@ -25,6 +29,13 @@ type (
 )
 
 func (r *CreateRequest) Validate() error {
+	// Set default configData schema.
+	if r.ConfigData.GetSchema() == nil {
+		if s := getConfigDataSchema(r.Category, r.Type); s != nil {
+			r.ConfigData.Schema = *s
+		}
+	}
```

Comment:
Why so many changes are needed around connectors? How do we use configdata schema? In my understanding, configData are basically map[string]string with sensitive bool.

Reply:
updated